openstack/cinder/templates/migration-job.yaml

```diff
@@ -10,15 +10,16 @@ spec:
   template:
     spec:
       restartPolicy: OnFailure
-      containers:
-        - name: cinder-migration
+{{ include "utils.proxysql.job_pod_settings" . | indent 6 }}
+      initContainers:
+        - name: dependencies
           image: {{required ".Values.global.registry is missing" .Values.global.registry}}/loci-cinder:{{.Values.imageVersionCinderApi | default .Values.imageVersion | required "Please set cinder.imageVersion or similar" }}
           imagePullPolicy: IfNotPresent
           command:
             - kubernetes-entrypoint
           env:
             - name: COMMAND
-              value: "cinder-manage db sync"
```

Comment:
how is the migration/upgrade from one version to another going to work without this call to the command that does the db sync/upgrade ?

Reply:
It is still called. I just split the container into two.
You are seeing just the init-container, which calls the `kubernetes-entrypoint` with command true.
The actual command is still executed in the normal container [here](https://github.com/sapcc/helm-charts/pull/3737/files/35c83eac79b1fd1c2eb2aff79c0b921e9f954eab#diff-4220c1b8749757535904420c740ccaad6f1aec9cd9a5c750f24773cc3cc4302eR39-R40) without kubernetes-entrypoint.

The main motivation was that this way I can easily introduce a shell script with the termination hook without having to create a configmap with a script in it.

It also has the benefit that aiting for the dependencies won't show up as restarts of the actual pod, but as an init-retry.